runtime/tests/checker/transactions_test.go

```diff
@@ -661,6 +661,90 @@ func TestCheckTransactionRoles(t *testing.T) {
 			nil,
 		)
 	})
+
+	t.Run("resource field", func(t *testing.T) {
+		test(t,
+			`
+              resource R {}
+
+              fun absorb(_ r: @R) {
+                  destroy r
+              }
+
+              transaction {
+
+                  role role1 {
+                      var r: @R
+
+                      prepare() {
+                          self.r <- create R()
+                      }
+                  }
+
+                  execute {
+                      absorb(<-self.role1.r)
```

Comment:
Can you also add a test to ensure that only the immediate members of a role can be moved, but not nested members?
i.e:
```cadence
absorb(<-self.role1.r.field)  // must give errors
```

Reply:
@SupunS Good idea! Added tests in [a91d406](https://github.com/onflow/cadence/pull/2262/commits/a91d4063d5eb9282966da44d6a7c737de94ac426) 

Also, along the way, improved the error message for the error, which helps users and allows better assertions in tests: [1a90380](https://github.com/onflow/cadence/pull/2262/commits/1a9038051d6850751994804c164c5440c5d58386)